src/main/scala/tech/cryptonomic/conseil/generic/chain/MultiFetchDecoding.scala

```diff
@@ -0,0 +1,209 @@
+package tech.cryptonomic.conseil.generic.chain
+
+import cats._
+import cats.data.Kleisli
+import cats.syntax.semigroupal._
+
+/** Contract for a generic combination of operations that get encoded data (usually json) from
+  * some resource (e.g. the tezos node), then converts that to a value, batching multiple requests together.
+  * The returned values are tupled with the corresponding input
+  * The interface assumes some kind of higher-order typed effect `F[_]` is part
+  * of the operation execution, and represents everything parametrically, to leave
+  * room for future evolutions of the code protocol.
+  * Thus the required implementation is in the form of a `cats.data.Kleisli` type,
+  * parametric on effect `F`, input `In`, output `Out`. As needed, the in/out types are
+  * wrapped in a `T[_]` type constructor that represents multiple values (e.g. a collection type).
+  *
+  * @tparam F the higher-order effect type (e.g. `Try`, `Future`, `IO`, `Either`)
+  * @tparam T a traversable type (used to wrap mutliple requests and corresponding return values)
+  */
+trait MultiFetchDecoding[F[_], T[_]] {
```

Comment:
As discussed on Riot, it would be good to rename the type parameters so they are more descriptive and the code is easy to read.

Reply:
Sure, will do